cmd/fs-v1-rwpool.go

```diff
@@ -95,6 +97,26 @@ func (fsi *fsIOPool) Open(path string) (*lock.RLockedFile, error) {
 
 		// Save new reader on the map.
 		fsi.Lock()
+		rlkFile, ok = fsi.readersMap[path]
+		if ok && rlkFile != nil {
+			// If the file is closed and not removed from map is a bug.
+			if rlkFile.IsClosed() {
+				// Log this as an error.
+				errorIf(errUnexpected, "Unexpected entry found on the map %s", path)
+
+				// Purge the cached lock path from map.
+				delete(fsi.readersMap, path)
+
+				// Save the newly acquired read locked file.
+				rlkFile = newRlkFile
+			} else {
+				// Increment the lock ref, since the file is not closed yet
+				// and caller requested to read the file again.
+				rlkFile.IncLockRef()
+				newRlkFile.Close()
+			}
+		}
+		// Save the rlkFile back on the map.
 		fsi.readersMap[path] = rlkFile
```

Comment:
if rlkFile was nil at line 100 then we would not be saving newRlkFIle to the map